src/Raven.Server/RavenServerStartup.cs

```diff
@@ -157,23 +159,90 @@ private async Task RequestHandler(HttpContext context)
                 context.Response.StatusCode = (int)HttpStatusCode.OK;
                 context.Response.Headers["Content-Type"] = "application/json; charset=utf-8";
                 context.Response.Headers[Constants.Headers.ServerVersion] = RavenVersionAttribute.Instance.AssemblyVersion;
+                // Check context url AND TW clients
+                bool isQueryBool = (TrafficWatchManager.HasRegisteredClients && IsQuery(context));
+
+                if (isQueryBool)
+                {
+                    // init new MS to save context.request.body
+                    MemoryStream memoryStreamForRequest = new MemoryStream();
+                    // copy stream
+                    await context.Request.Body.CopyToAsync(memoryStreamForRequest);
```

Comment:
I don't like this approach. Better to pass a flag to the actions and have them construct their own custom data, instead.